src/frontend/src/flows/manage/deviceSettings.ts

```diff
@@ -99,20 +97,32 @@ const isProtected = (device: DeviceData): boolean =>
 const isRecovery = (device: DeviceData): boolean =>
   "recovery" in device.purpose;
 
+export const deviceSettingsPage = (
+  props: Parameters<typeof deviceSettingsTemplate>[0],
+  container?: HTMLElement
+): void => {
+  const contain =
+    container ?? (document.getElementById("pageContent") as HTMLElement);
+  render(deviceSettingsTemplate(props), contain);
+};
+
 // Get the list of devices from canister and actually display the page
 export const deviceSettings = async (
   userNumber: bigint,
   connection: AuthenticatedConnection,
   device: DeviceData,
   isOnlyDevice: boolean
 ): Promise<void> => {
-  const container = document.getElementById("pageContent") as HTMLElement;
-
   return new Promise((resolve) => {
-    render(
-      pageContent(userNumber, connection, device, isOnlyDevice, resolve),
-      container
-    );
+    deviceSettingsPage({
+      device,
+      isOnlyDevice,
+      back: resolve,
```

Comment:
just to be sure, `resolve` is not a reserved keyword right?

Reply:
it's not! I think you can name the callback anything you want.